service/controller/resource/nodepool/deployment.go

```diff
@@ -104,6 +104,10 @@ func (r Resource) newDeployment(ctx context.Context, storageAccountsClient *stor
 func (r Resource) getSubnetName(azureMachinePool *capzexpv1alpha3.AzureMachinePool, azureCluster *capzv1alpha3.AzureCluster) (string, string, error) {
 	for _, subnet := range azureCluster.Spec.NetworkSpec.Subnets {
 		if azureMachinePool.Name == subnet.Name {
+			if subnet.ID == "" {
+				return azureCluster.Spec.NetworkSpec.Vnet.Name, subnet.Name, microerror.Maskf(subnetNotReadyError, fmt.Sprintf("Subnet %#q ID field is empty, which means the Subnet is not Ready", subnet.Name))
```

Comment:
When returning an error we should return empty values for other fields. Is there something in the logic where we need the vnet and subnet names with the error?

Reply:
Nope. I'll change it.